cmd/mt-parrot/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out"
+	"github.com/grafana/metrictank/cmd/mt-fakemetrics/out/gnet"
+	"github.com/grafana/metrictank/logger"
+	"github.com/grafana/metrictank/schema"
+	"github.com/raintank/met/statsd"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+var (
+	gatewayAddress        string
+	gatewayKey            string
+	orgId                 int
+	partitionCount        int32
+	partitionMethodString string
+	testMetricsInterval   time.Duration
+	queryInterval         time.Duration
+	logLevel              string
+
+	partitionMethod schema.PartitionByMethod
+	gateway         out.Out
+)
+
+func init() {
+	parrotCmd.Flags().StringVar(&gatewayAddress, "gateway-address", "http://localhost:6059", "the url of the metrics gateway")
+	parrotCmd.Flags().StringVar(&gatewayKey, "gateway-key", "", "the bearer token to include with gateway requests")
+	parrotCmd.Flags().IntVar(&orgId, "org-id", 1, "org id to publish parrot metrics to")
+	parrotCmd.Flags().Int32Var(&partitionCount, "partition-count", 8, "number of kafka partitions in use")
+	parrotCmd.Flags().StringVar(&partitionMethodString, "partition-method", "bySeries", "the partition method in use on the gateway, must be one of bySeries|bySeriesWithTags|bySeriesWithTagsFnv")
+	parrotCmd.Flags().DurationVar(&testMetricsInterval, "test-metrics-interval", 10*time.Second, "interval to send test metrics")
```

Comment:
can you confirm what happens when you pass invalid options for this flag?
we've had some issues in the past with duration flags silently ignoring errors.  (you can look at other code elsewhere or try to pull up old PR's to see how we handled it). but maybe with cobra it's not an issue :?

Reply:
```$ ./mt-parrot --test-metrics-interval=20
Error: invalid argument "20" for "--test-metrics-interval" flag: time: missing unit in duration 20
Usage:
  mt-parrot [flags]

Flags:
      --gateway-address string           the url of the metrics gateway (default "http://localhost:6059")
      --gateway-key string               the bearer token to include with gateway requests
  -h, --help                             help for mt-parrot
      --log-level string                 log level. panic|fatal|error|warning|info|debug (default "info")
      --org-id int                       org id to publish parrot metrics to (default 1)
      --partition-count int32            number of kafka partitions in use (default 8)
      --partition-method string          the partition method in use on the gateway, must be one of bySeries|bySeriesWithTags|bySeriesWithTagsFnv (default "bySeries")
      --query-interval duration          interval to query to validate metrics (default 10s)
      --test-metrics-interval duration   interval to send test metrics (default 10s)

2020-02-26 14:36:11.843 [FATAL] invalid argument "20" for "--test-metrics-interval" flag: time: missing unit in duration 20
```

Reply:
great thx